xija/model.py

```diff
@@ -692,8 +687,8 @@ def calc(self):
                                      len(self.tmal_ints), dt, mvals,
                                      tmal_ints, tmal_floats)
 
-        # hackish fix to ensure last value is computed
-        self.mvals[:, -1] = self.mvals[:, -2]
+        # hackish fix to ensure last value is a computed value for predicted components
+        self.mvals[:self.n_preds, -1] = self.mvals[:self.n_preds, -2]
```

Comment:
This is the meat of the PR.  The rest is just PEP8 improvements.

Reply:
Looks good.